app/src/main/java/com/example/microtemp/microblog/RegisterActivity.java

```diff
@@ -1,13 +1,131 @@
+        password = findViewById(R.id.password) ;
+        surname=findViewById(R.id.Surname);
+        retype_password=findViewById(R.id.retype_password);
+        name=findViewById(R.id.name);
+        backBtn = findViewById(R.id.back_button);
+
+        backBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(RegisterActivity.this, LoginActivity.class);
+                startActivity(intent);
+            }
+        });
+
+        registerBtn = findViewById(R.id.register_button);
+        registerBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (check() == true) {
+                    RegisterRequestBody requestBody = RegisterRequestBody.builder()
+                            .email(email.getText().toString())
+                            .password(password.getText().toString())
+                            .firstName(name.getText().toString())
+                            .lastName(surname.getText().toString())
+                            .admin(false)
+                            .build();
+                    HttpRequestData<RegisterRequestBody> requestData = HttpRequestData.<RegisterRequestBody>builder()
+                            .requestBody(requestBody)
+                            .method(HttpRequestMethods.POST)
+                            .build();
+                    new RegisterRequestHandler() {
+                        @Override
+                        protected void onPostExecute(RegisterResponse result) {
+                            if (result.getHttpStatusCode() == 400) {
+                                Toast.makeText(registerBtn.getContext(),
+                                        "Nie można zarejestrować nowego użytkownika, ponieważ nie ma dostępu do internetu",
+                                        Toast.LENGTH_LONG).show();
+                            } else if (result.getHttpStatusCode() == 201) {
+                                Intent intent = new Intent(RegisterActivity.this, LoginActivity.class);
+                                startActivity(intent);
+                                Toast.makeText(registerBtn.getContext(),
+                                        "Uzytkownik zarejestrowany",
+                                        Toast.LENGTH_LONG).show();
+                            } else {
+                                Intent intent = new Intent(RegisterActivity.this, MainActivity.class);
+                                intent.putExtra("response", result);
+                                startActivity(intent);
+                                Log.d(REG_TAG, "Result: " + result);
+                            }
+                        }
+                    }.execute(requestData);
+                }
+            } });
+
+    }
+    public boolean check()
```

Comment:
Od tego momentu zadziało się coś złego z formatowaniem, brakuje wcięć, klamry są w osobnych liniach, często brakuje spacji, ogólnie można by to trochę zrefaktoryzować